### DISCUSSION TITLE
실행 컨텍스트란 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#69] [실행컨텍스트]
- 실행 컨텍스트란 무엇인가요?
  - 실행 컨텍스트 스택이란 무엇인가요?
  - 실행 컨텍스트의 구성 요소에는 무엇이 있나요? 렉시컬 환경의 구성 요소까지 설명해주세요.

**Comment by publdaze** [comment]
> - 실행 컨텍스트란 무엇인가요?
> - 실행 컨텍스트 스택이란 무엇인가요?
- 소스 코드를 실행하는데 필요한 환경을 제공하고, 코드 실행 결과를 실제로 관리하는 영역
- 함수가 호출되면 실행 컨텍스트가 생성되어 실행 순서를 관리하는 실행 컨텍스트에 쌓이게 되고, 실행이 끝나면 실행 컨텍스트에서 제거가 되는 후입선출 방식으로 동작을 하고, 실행 컨텍스트가 생성이 되면서 평가단계에서 생성이 되는 렉시컬 환경을 통해 식별자와 스코프 등을 관리할 수 있습니다.

**Comment by pipisebastian** [comment]
- 실행 컨텍스트란, 함수가 실행될 때, 해당 함수와 관련된 환경 정보를 가지는 객체입니다. 
- 실행 컨텍스트는 스택 자료구조로 관리되는데, 이를 실행컨텍스트 스택이라 합니다. 이를 콜스택이라고도 합니다. 전역 컨텍스트가 제일 먼저 쌓이고, 제일 마지막에 빠져나갑니다.
- 자바스크립트는 하나의 콜스택을 갖는 싱글 스레드 언어이기 때문에, 한번의 하나의 함수만 실행할 수 있습니다.
